Update fabric mod to 1.21.9

amiralimollaei opened the pull request:
Updated the fabric version of the mod to 1.21.9, the required bumping versions of the dependencies and minimal changes to a few mixins, as well as renaming some function interfaces to match the changes made in the Minecraft update, I made sure the functionality states the same.

amiralimollaei committed:
Update fabric mod to 1.21.9

squeek502 commented:
Nice work, thanks!

According to this blog post, `DebugHudMixin` should be able to be deleted: https://fabricmc.net/2025/09/23/1219.html#debug-text-api

If you want to take a crack at implementing that, go right ahead. Otherwise, I'll get to it in a few days.

amiralimollaei committed:
Properly implement `DebugHudEntry`

amiralimollaei committed:
Properly implement `DebugHudEntry`

Also added `MixinDebugHudEntries` to put the `DebugInfoHudEntry` instance
inside `PROFILES` to show up in the default debug profile, and not in any
other debug profile

Without `MixinDebugHudEntries` it does not show up at all.

amiralimollaei commented:
Thanks, I didn't catch that in the blog post, sorry, now it uses `DebugHudEntry` but it still needed a new mixin `MixinDebugHudEntries` to add the entry to the default debug profile, otherwise it would not render, even though it was registered.

I've added it to the position section of the debug hud, alongside other position and biome information.

amiralimollaei committed:
rename `MixinDebugHudEntries` to `DebugHudEntriesMixin`

squeek502 committed:
Allow debug info visibility to be fully controlled by F3 + F6 config

Removes the `showFoodDebugInfo` config option in favor of letting vanilla Minecraft handle it.

squeek502 commented:
I'm cool with the debug info not being visible by default and letting the vanilla debug info config system handle the visibility, so I went ahead and made that change. Everything else looks good.

Thanks again, really appreciate you working on this!

amiralimollaei commented:
Great! and thank you for this amazing mod
